src/app/EventManagement.h

```diff
@@ -188,6 +188,25 @@ enum class EventManagementStates
     Shutdown   = 3  // Not capable of performing any logging operation
 };
 
+/**
+ * @brief
+ *  Internal structure for traversing events.
+ */
+struct EventEnvelopeContext
```

Comment:
Do we need to move this inside the header?

Reply:
no need now, thanks, resovled.